Decription: fix double free in pam_smbpass.
 Since https://bugzilla.samba.org/show_bug.cgi?id=11066 got fixed, the
 ret_data_cleanup() callback function should be freeing memory, and doing
 so also in pam_sm_setcred() causes a double-free.
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/samba/+bug/1515207
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=799840

Index: samba-4.3.7/source3/pam_smbpass/pam_smb_auth.c
===================================================================
--- samba-4.3.7.orig/source3/pam_smbpass/pam_smb_auth.c	2015-07-21 05:47:49.000000000 -0400
+++ samba-4.3.7/source3/pam_smbpass/pam_smb_auth.c	2016-04-07 09:19:14.040332123 -0400
@@ -188,7 +188,6 @@
 	_pam_get_data(pamh, "smb_setcred_return", &pretval);
 	if(pretval) {
 		retval = *pretval;
-		SAFE_FREE(pretval);
 	}
 	pam_set_data(pamh, "smb_setcred_return", NULL, NULL);
 
